testdata/issue5.go

```diff
@@ -13,6 +13,7 @@ func Gen() S {
 // Gen().Layout() has another inlined function in it, Gen().
 
 //gcassert:inline
+//go:noinline
```

Comment:
I had to add `go:noinline`, otherwise the test fails; it seems inlining may have improved in this case; from `go build`,

```
testdata/issue5.go:16:6: can inline S.Layout with cost 1 as: method(S) func() { select statement }
testdata/issue5.go:3:6: can inline bar with cost 10 as: func() { S.Layout(.autotmp_0) }
testdata/issue5.go:4:5: inlining call to Gen
testdata/issue5.go:4:14: inlining call to S.Layout
```